### PR TITLE
36 create new project button on the projects page

### LINKED ISSUE #36: Create new Project Button on the projects Page
Drawer component should open a form the side to create a new Project

### PR DESCRIPTION
**Pull request by MateWagner**

### CODE REVIEW

File: src/frontend/src/pages/projects/NewProjectDrawer.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { Button, Drawer, Form, Input, message } from "antd";
+import TextArea from "antd/es/input/TextArea";
+
+const NewProjectDrawer = ({ onClose, open, addNewProject }) => {
+  const [form] = Form.useForm();
+
+  const onFinish = async (values) => {
+    console.log(form);
```
Comment (tuszmak) [inline src/frontend/src/pages/projects/NewProjectDrawer.js]: remove this console log

File: src/frontend/src/pages/projects/ProjectList.js [hunk 2]
```diff
@@ -1,27 +1,61 @@
...
-      render: (text, record) => <Link to={`/project/${record.id}`}>{text}</Link>,
-    }
+      title: "Name",
+      dataIndex: "name",
+      key: "name",
+      render: (text, record) => (
+        <Link to={`/project/${record.id}`}>{text}</Link>
+      ),
+    },
   ];
```
Comment (tuszmak) [inline src/frontend/src/pages/projects/ProjectList.js]: Should we display question description as well? It shouldn't be the whole description, but like a 50 letter version. We already have the data for it, if we want to.

**Review by tuszmak — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/projects/NewProjectDrawer.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const option = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(values),
+    };
+
+    const result = await fetch("/api/v1/project", option);
+    if (result.status === 400) {
```
Comment (gerigol) [inline src/frontend/src/pages/projects/NewProjectDrawer.js]: Checking if the response was succefull is more flxible with the result.ok property ( https://developer.mozilla.org/en-US/docs/Web/API/Response/ok ) -> if (!result.ok)

File: src/frontend/src/pages/projects/NewProjectDrawer.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        {
+          name: "name",
+          errors: [error.message],
+        },
+      ]);
+      return;
+    }
+    if (!result.ok) {
+      const ERROR_MSG_DURATION = 3;
+      message.error("Problem with the Server", ERROR_MSG_DURATION);
```
Comment (gerigol) [inline src/frontend/src/pages/projects/NewProjectDrawer.js]: The error message should be provided dinamically:  const errorMessage = await error.message;
message.error(errorMessage, ERROR_MSG_DURATION);

File: src/frontend/src/pages/projects/NewProjectDrawer.js (hunk 3 above)
Comment (MateWagner) [reply src/frontend/src/pages/projects/NewProjectDrawer.js]: This is a special case to give an error to the form field the rest is catch under

File: src/frontend/src/pages/projects/NewProjectDrawer.js (hunk 4 above)
Comment (MateWagner) [reply src/frontend/src/pages/projects/NewProjectDrawer.js]: i don't think await need it's a antd component https://ant.design/components/message

**Pull request merged**